edx/blockstore
# Make blockstore an installable package

## Description

This PR pull out the following changes from https://github.com/edx/blockstore/pull/97:
* Move apps/api to apps/rest_api.
* Copy the latest code in `edx-platform/openedx/core/lib/blockstore_api` to `blockstore/apps/api`. This has been done to make the review of https://github.com/edx/blockstore/pull/97 easier since we will get a useful diff of changes made to each of the API methods.
* Makes the apps into an installable package.

It also disables the migration code which changes the MySQL character set to `utfmb4` to prevent accidentally running it against an edxapp database.

## Ticket

* [FAL-170](https://tasks.opencraft.com/browse/FAL-170)

## Author Comments, Concerns, and Open Questions

## Test Instructions

* Checkout branch.
* In the Open edX devstack run:
```
make dev.shell.mysql57
mysql -u root
DROP DATABASE blockstore_db;
```

* Start and provision the blockstore docker container.
* Run tests with `make test`.
* Start the blockstore django server and check the API is working correctly using the Libraries UI or LabXchange devstack.

Add `'blockstore.apps.bundles.apps.BundlesConfig',` to `lms/envs/common.py::INSTALLED_APPS`.

Install the package in LMS:
```
make lms-shell
pip install -e /edx/src/blockstore
./manage.py lms migrate
```

Check the API methods can be called:
```
./manage.py lms shell
from blockstore.api import *
create_collection("Collection") # This should fail with requests.exceptions.ConnectionError
```

## TODOs

None
